app/src/main/fetch/queue.ts

```diff
@@ -0,0 +1,86 @@
+import Queue from "better-queue";
+
+import { DB } from "../database";
+import { FetchStatus } from "../../types";
+
+export type ItemFetchTask = {
+  id: string;
+};
+
+export function createQueue(db: DB): Queue {
+  const q: Queue = new Queue(
+    function (task: ItemFetchTask, onComplete) {
+      try {
+        fetchDownload(task, db);
+        onComplete();
+      } catch (e) {
+        console.log("Error executing fetch download task: ", task, e);
+        db.failItem(task.id);
+        throw e;
+      }
+    },
+    {
+      batchSize: 1,
+      maxTimeout: 30_000,
```

Comment:
I know this is just a provisional value for setting up the queue, and it may well be a reasonable default for text messages/replies.  For files, it may be worth having a design discussion about how to set a [Tor-realistic timeout][_get_realistic_timeout] around the proxy's treaming semantics.


[_get_realistic_timeout]: https://github.com/freedomofpress/securedrop-client/blob/a6ff9346f1877ecb4d62010a35a6bd38f6ea2b0a/client/securedrop_client/api_jobs/downloads.py#L62-L93